cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/ExecutorStatusTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource.cruisecontrol;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.strimzi.operator.common.model.cruisecontrol.CruiseControlExecutorState;
+import org.junit.jupiter.api.Test;
+
+import static io.strimzi.operator.cluster.operator.resource.cruisecontrol.ExecutorStatus.FINISHED_DATA_MOVEMENT_KEY;
+import static io.strimzi.operator.cluster.operator.resource.cruisecontrol.ExecutorStatus.STATE_KEY;
+import static io.strimzi.operator.cluster.operator.resource.cruisecontrol.ExecutorStatus.TOTAL_DATA_TO_MOVE_KEY;
+import static io.strimzi.operator.cluster.operator.resource.cruisecontrol.ExecutorStatus.TRIGGERED_TASK_REASON_KEY;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ExecutorStatusTest {
+    private static final String DEFAULT_STATE = CruiseControlExecutorState.INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS.toString();
+    private static final String DEFAULT_FINISHED_DATA_MOVEMENT = "50";
+    private static final String DEFAULT_TOTAL_DATA_TO_MOVE = "1000";
+    private static final String DEFAULT_TRIGGERED_TASK_REASON = "No reason provided (Client: 172.17.0.1, Date: 2024-11-15T19:41:27Z)";
+
+    public static ObjectNode createExecutorStatusJson(String state, String finishedDataMovement, String totalDataToMove, String triggeredTaskReason) {
```

Comment:
What about changing the type of `state` to `CruiseControlExecutorState`? That way you don't have to call `toString()` in all the callers.

Reply:
Fixed.